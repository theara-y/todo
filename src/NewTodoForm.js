import { useState } from "react";

const NewTodoForm = ({addTodo}) => {
    const [text, setText] = useState("");

    const handleChange = (e) => {
        setText(e.target.value);
    }

    const handleSubmit = (e) => {
        e.preventDefault();
        if (text != "") {
            addTodo({text, done: false});
            setText("");
        }
    }

    return (
        <form onSubmit={handleSubmit}>
            <input type="text" value={text} onChange={handleChange} placeholder="Add Todo..."/>
            <button>Add Todo</button>
        </form>
    );
};

export default NewTodoForm;