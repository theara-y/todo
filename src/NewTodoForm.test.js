import { render } from "@testing-library/react";
import NewTodoForm from "./NewTodoForm";

it("renders", () => {
    render(<NewTodoForm/>);
});

it("matches snapshots", () => {
    const { asFragment } = render(<NewTodoForm/>)
    expect(asFragment()).toMatchSnapshot();
})