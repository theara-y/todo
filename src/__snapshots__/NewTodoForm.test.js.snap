// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshots 1`] = `
<DocumentFragment>
  <form>
    <input
      placeholder="Add Todo..."
      type="text"
      value=""
    />
    <button>
      Add Todo
    </button>
  </form>
</DocumentFragment>
`;
